static/canvas.js

```diff
@@ -200,45 +187,48 @@ class Canvas {
 
```

Comment:
Not sure why I had myState.valid = false in the line above, but it should take care of redrawing shapes to their proper location. Still, I think you need to update this.boxes to accommodate the current frame.